src/NuGet.Clients/NuGet.PackageManagement.UI/Models/Package/EmbeddedResourcesCapability.cs

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+#nullable enable
+
+using System;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using NuGet.VisualStudio.Internal.Contracts;
+
+namespace NuGet.PackageManagement.UI.Models.Package
+{
+    internal class EmbeddedResourcesCapability : IEmbeddedResources
+    {
+        private INuGetPackageFileService _nugetPackageFileService;
+        private PackageModel _package;
+
+        public EmbeddedResourcesCapability(INuGetPackageFileService nugetPackageFileService, PackageModel package, Uri? iconUri, Uri? licenseUri, Uri? readmeUri)
+        {
+            _nugetPackageFileService = nugetPackageFileService ?? throw new ArgumentNullException(nameof(nugetPackageFileService));
+            _package = package ?? throw new ArgumentNullException(nameof(package));
+            IconUri = iconUri;
+            LicenseUri = licenseUri;
+            ReadmeUri = readmeUri;
+        }
+
+        public Uri? IconUri { get; }
+
+        public Uri? LicenseUri { get; }
+
+        public Uri? ReadmeUri { get; }
+
+        public async Task<Stream?> GetIconAsync(CancellationToken cancellationToken)
+        {
+            return await _nugetPackageFileService.GetPackageIconAsync(_package.Identity, cancellationToken);
```

Comment:
Since we don't really need to await the operation in this method to do anything else to it, what if we just return the task and let the caller decide when to await it?

```suggestion
            return _nugetPackageFileService.GetPackageIconAsync(_package.Identity, cancellationToken);
```

Reply:
Removed the await required switching to value task, which seems fine.